docs/rollup_internals.md

```diff
@@ -0,0 +1,52 @@
+# Technical notes on `rollup`
+## About
+Most aggregates in Toolkit have an associated `rollup` aggregate that aggregates multiple aggregates covering various ranges of time and combines them into one merged aggregate.
+
+Most `rollup` aggregates require that each aggregate to be rolled up cover a non-overlapping period of time.
```

Comment:
I think this is still a minority of our aggregates that have this limitation on aggregation.  It would be nice to make it more explicit that this approach is specifically for aggregates where we require time-ordered input, and that other cases should continue to implement fully parallel rollups.